src/k2/cmd/ycsb/transactions.h

```diff
@@ -0,0 +1,323 @@
+                    });
+                })
+                .then([this] () {
+                    K2LOG_D(log::ycsb, "Scan succeeded");
+                    return seastar::make_ready_future();
+                });
+            });
+        });
+    }
+
+    seastar::future<> insertOperation(){
+
+        if(_requestDistName()!="latest") {
+            K2LOG_D(log::ycsb, "Insert operation started for keyid {}", _keyid);
+            YCSBData row(_keyid, _random); // generate row
+            return writeRow(row, _txn).discard_result();
+        }
+
+        _keyid = _requestDist->getMaxValue()+1;
+        YCSBData row(_keyid,_random); // key is max known (latest) key + 1
+
+        //handle latest distribution separately to identify insert fails and increment latest known record max key value
+        K2LOG_D(log::ycsb, "Insert operation started for keyid {}", _keyid);
+        return writeRow(row, _txn, false, ExistencePrecondition::Exists) // if record with given key already exists must return error so we set precondition to Exists
+            .then_wrapped([this] (auto&& fut) {
+                if (fut.failed()) {
+                    fut.ignore_ready_future();
+                    return seastar::make_exception_future<>(std::runtime_error("Insert failed!"));
+                }
+
+                WriteResult result = fut.get0();
+                if (result.status.is2xxOK() || result.status.code == 403) { // key inserted or already exists
+                    if(_requestDist->getMaxValue() < _keyid) { // update bounds of distribution
+                        _requestDist->updateBounds(0,_keyid);
+                    }
+                    if(result.status.code == 403) { // insert missed because record exists
+                        _insertMissesLatest++; // increment misses counter
+                    }
+                }
+
+                return seastar::make_ready_future();
+            });
+    }
+
+    seastar::future<> deleteOperation(){
+        K2LOG_D(log::ycsb, "Delete operation started for keyid {}", _keyid);
+        YCSBData row(_keyid); // generate row
+        return writeRow(row, _txn, true).discard_result();
+    }
+
+    Query _query_scan;
+    K2TxnHandle _txn;
+    RandomContext& _random;
+    K23SIClient& _client;
+    bool _failed;
+    YCSBData _data; // row read from read operation
+    uint64_t _keyid; // keyid for current op
+    std::vector<uint64_t> _txnkeyids; // keyids used in this txn stored for retries
+    std::vector<Operation> _ops; // ops used in this txn stored for retries
+    std::vector<YCSBData> _scanResult; // rows read from scan operation
+    std::unique_ptr<RandomGenerator>& _requestDist; // Request distribution for selecting keys
+    std::unique_ptr<RandomGenerator>& _scanLengthDist; // Request distribution for selecting length of scan
+    uint64_t& _insertMissesLatest; // number of inserts missed when request distribution is Latest distribution
```

Comment:
Can you make this not a reference the same way you changed the ops vector?

Reply:
done